### PR TITLE
Audit API and doc for Delta Standalone Reader

### PR DESCRIPTION
**Pull request by zsxwing**
Review public APIs and make minor changes:
- Remove unused or redundant APIs.
- Add missing equals and hashCode methods.
- Fix the null values for primitive types in RecordValue.
- Minor doc changes.

### CODE REVIEW

File: standalone/src/main/java/io/delta/standalone/DeltaLog.java [hunk 1]
```diff
@@ -41,63 +57,43 @@
...
-     */
-    static DeltaLog forTable(Configuration hadoopConf, String dataPath) {
-        return DeltaLogImpl.forTable(hadoopConf, dataPath);
-    }
+    /** @return the path of the Delta table. */
+    Path getPath();
 
     /**
-     * Helper for creating a log when it is stored at the root of the data.
+     * Create a {@link DeltaLog} instance representing the table located at the provide {@code path}.
```
Comment (scottsand-db) [inline standalone/src/main/java/io/delta/standalone/DeltaLog.java:79]: nit: provided

File: standalone/src/main/java/io/delta/standalone/Snapshot.java [hunk 2]
```diff
@@ -23,46 +37,20 @@
...
-    int getNumOfFiles();
-
-    /**
-     * Creates a new {@code CloseableIterator<RowRecord>} which can iterate over data located in this
-     * snapshot's {@code DeltaLog.dataPath}. Iterates file by file, row by row.
-     * <p>
-     * This does not iterate over snapshot metadata/log files. To do that, use {@code getAllFiles}.
-     * <p>
-     * Provides no iteration ordering guarantee among data files.
+     * Creates a {@link CloseableIterator} which can iterate over data belonging to the version of
```
Comment (scottsand-db) [inline standalone/src/main/java/io/delta/standalone/Snapshot.java:50]: nit: belonging to this snapshot version? belonging to this snapshot?
version is just an integer; data doesn't belong to an integer

File: standalone/src/main/java/io/delta/standalone/actions/AddFile.java [hunk 3]
```diff
@@ -33,24 +52,18 @@ public AddFile(String path, Map<String, String> partitionValues, long size,
     }
 
     /**
-     * @return the relative path, from the root of the table, to the data file
-     *         that should be added to the table
+     * @return the relative path or the absolute path that should be added to the table. If it's a
+     *         relate path, it's relative to the root of the table. Note: the path is encoded and
```
Comment (scottsand-db) [inline standalone/src/main/java/io/delta/standalone/actions/AddFile.java:56]: nit: relative path, spelling mistake

File: standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala [hunk 4]
```diff
@@ -69,7 +67,7 @@ private[internal] class DeltaLogImpl private(
   }
 }
 
-private[standalone] object DeltaLogImpl {
+private[internal] object DeltaLogImpl {
```
Comment (scottsand-db) [inline standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala:70]: DeltaLog.java has this code:
```
static DeltaLog forTable(Configuration hadoopConf, String path) {
        return DeltaLogImpl.forTable(hadoopConf, path);
    }
```
Dont we want DeltaLogImpl.forTable to be `private[standalone]` to indicate that it is okay for DeltaLog.java to use this method?

File: standalone/src/main/scala/io/delta/standalone/internal/actions/actions.scala [hunk 5]
```diff
@@ -243,7 +243,7 @@ private[internal] case class SingleAction(
 }
 
 /** Serializes Maps containing JSON strings without extra escaping. */
-class JsonMapSerializer extends JsonSerializer[Map[String, String]] {
+private[internal] class JsonMapSerializer extends JsonSerializer[Map[String, String]] {
```
Comment (scottsand-db) [inline standalone/src/main/scala/io/delta/standalone/internal/actions/actions.scala:246]: good catch, thanks

**Review by scottsand-db — APPROVED** [review]
great job ryan. thanks for this. small nit comments.

File: standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala (hunk 4 above)
Comment (zsxwing) [reply standalone/src/main/scala/io/delta/standalone/internal/DeltaLogImpl.scala]: Good point. I forgot it.

**Pull request merged**